sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import division
+
+import numpy as np
+from numpy.testing import run_module_suite
+from scipy.sparse import csr_matrix
+
+from sklearn.utils.testing import (assert_array_equal, assert_almost_equal,
+                                   assert_false, assert_true, assert_raises)
+from sklearn.feature_selection.mutual_info_ import (
+    mutual_info_regression, mutual_info_classif, _compute_mi)
+
+
+class TestMIComputation(object):
+    def test_dd(self):
+        # In discrete case computations are straightforward and can be done
+        # by hand on given vectors.
+        x = np.array([0, 1, 1, 0, 0])
+        y = np.array([1, 0, 0, 0, 1])
+
+        H_x = H_y = -(3/5) * np.log(3/5) - (2/5) * np.log(2/5)
+        H_xy = -1/5 * np.log(1/5) - 2/5 * np.log(2/5) - 2/5 * np.log(2/5)
+        I_xy = H_x + H_y - H_xy
+
+        assert_almost_equal(_compute_mi(x,  y, True, True), I_xy)
+
+    def test_cc(self):
+        # For two continuous variables a good approach is to test on bivariate
+        # normal distribution, where mutual information is known.
+
+        # Mean of the distribution, irrelevant for mutual information.
+        mean = np.zeros(2)
+
+        # Setup covariance matrix with correlation coeff. equal 0.5.
+        sigma_1 = 1
+        sigma_2 = 10
+        corr = 0.5
+        cov = np.array([
+            [sigma_1**2, corr * sigma_1 * sigma_2],
+            [corr * sigma_1 * sigma_2, sigma_2**2]
+        ])
+
+        # True theoretical mutual information.
+        I_theory = (np.log(sigma_1) + np.log(sigma_2) -
+                    0.5 * np.log(np.linalg.det(cov)))
+
+        np.random.seed(0)
+        Z = np.random.multivariate_normal(mean, cov, size=1000)
+
+        x, y = Z[:, 0], Z[:, 1]
+
+        # Theory and computed values won't be very close, assert that relative
+        # error is less than 10%.
```

Comment:
I'll check how much I can reduce it.

I don't think it's too much, because MI is a non-parametric method. My intuition is that those methods require more data for a stable estimation.